trigger_pentest.sh

```diff
@@ -1,14 +1,75 @@
-
 URL=https://api.getastra.com/webhooks/integrations/ci-cd
+
 ASTRA_SCAN_TYPE="${ASTRA_SCAN_TYPE:-lightning}"
-status_code=$(curl -s -o response.txt -w "%{http_code}" --user-agent "Astra Pentest Trigger Script/1.0" --header "Content-Type: application/json" --request POST --data "{\"accessToken\":\"$ASTRA_ACCESS_TOKEN\",\"projectId\":\"$ASTRA_PROJECT_ID\", \"mode\":\"$ASTRA_AUDIT_MODE\", \"automatedScanType\":\"$ASTRA_SCAN_TYPE\"}" $URL)
-if [[ "$status_code" == "200" ]] ; then
+ASTRA_JOB_EXIT_STRATEGY="${ASTRA_JOB_EXIT_STRATEGY:-always_pass}"
+ASTRA_JOB_EXIT_REFETCH_INTERVAL="${ASTRA_JOB_EXIT_REFETCH_INTERVAL:-120}"
+ASTRA_JOB_EXIT_REFETCH_MAX_RETRIES="${ASTRA_JOB_EXIT_REFETCH_MAX_RETRIES:-10}"
+ASTRA_JOB_EXIT_CRITERION='{"jobExitCriterion": "severityCount[\"high\"] > 10 or severityCount[\"critical\"] > 10"}'
+
+# Make the first API call
+status_code=$(curl -s -o response.txt -w "%{http_code}" --user-agent "Astra Pentest Trigger Script/1.0" --header "Content-Type: application/json" --request POST --data "{\"accessToken\":\"$ASTRA_ACCESS_TOKEN\",\"projectId\":\"$ASTRA_PROJECT_ID\", \"mode\":\"$ASTRA_AUDIT_MODE\", \"automatedScanType\":\"$ASTRA_SCAN_TYPE\", \"jobExitStratgey\":\"$ASTRA_JOB_EXIT_STRATEGY\", \"jobExitCriterion\":\"$ASTRA_JOB_EXIT_CRITERION\", \"jobExitRefetchInterval\":\"$ASTRA_JOB_EXIT_REFETCH_INTERVAL\", \"jobExitPrefetchMaxRetries\":\"$ASTRA_JOB_EXIT_REFETCH_MAX_RETRIES\", \"targetScopeUri\":\"$ASTRA_TARGET_SCOPE_URI\"}" $URL)
+
+# Check the status code of the first API call
+if [[ "$status_code" == "200" ]]; then
   echo "✅ Astra pentest was successfully started."
   cat response.txt
-elif [[ "$status_code" == "422" ]] ; then
+elif [[ "$status_code" == "422" ]]; then
   echo "🟡 Cannot start an audit, because an audit might be underway."
   cat response.txt
 else
   echo "⛔ Failed to start pentest."
   exit 1
 fi
+
+# Store the ID returned from the first API call
+audit_id=$(jq -r '.auditId' response.txt)
+
+# Define the JSON data for the second API call
+json_data="{\"jobExitCriterion\": $ASTRA_JOB_EXIT_CRITERION}"
+
+# Define the URL for the second API call
+scan_status_url="https://api.getastra.com/api/vapt/audits/$audit_id/cicd-scan-status"
+
+# Loop for ASTRA_JOB_EXIT_REFETCH_MAX_RETRIES number of times
+for ((retry=0; retry<ASTRA_JOB_EXIT_REFETCH_MAX_RETRIES; retry++)); do
+  # Call the second API and store the response
+  scan_status=$(curl -s -o scan_status_response.txt -w "%{http_code}" --user-agent "Astra Pentest Trigger Script/1.0" --header "Content-Type: application/json" --request GET --data "$json_data" $scan_status_url)
+
+  # Parse the response to extract audit progress
+  audit_progress=$(jq -r '.auditProgress' scan_status_response.txt)
```

Comment:
We should check the HTTP status code before checking for audit progress. If its a failure code, then the audit progress would not be returned and might cause errors.

Reply:
So flow would be:
1. Check for HTTP status code. Exit on failure.
2. Check for exit criterion. If it's true then fail the pipeline and exit.
3. Check for audit status. If it's not in pending/automated state then pass the job. If it's still in pending/automated state then retry after the set itnerval